packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1524,7 +1524,7 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
           plannedInitialRoutes.add(_routeNamed<dynamic>(routeName, allowNull: true, arguments: null));
         }
       }
-      if (plannedInitialRoutes.contains(null)) {
+      if (_shouldAbandonInitialRoute(plannedInitialRoutes)) {
         assert(() {
           FlutterError.reportError(
             FlutterErrorDetails(
```

Comment:
The documentation on various `initialRoute` arguments in Flutter (e.g. on Navigator, MaterialApp, etc) needs to be updated to reflect the new behavior.